public/externalLibs/visualizer/visualizer.js

```diff
@@ -809,77 +856,112 @@
-     * for height.
-     * In practice, likely due to browser auto-scaling, for large drawings this results in
-     * some of the drawing being cut off. Hence the width and height formulas used are approximations.
-     */
-    let stage = new Konva.Stage({
-      width: findListWidth(xs) * 60 + 60,
-      height: findListHeight(xs) * 60 + 100,
-      container: 'list-visualizer-container'
-    });
     minLeft = 500;
     nodelist = [];
     fnNodeList = [];
     nodeLabel = 0;
-    // hides all other layers
-    for (let i = 0; i < layerList.length; i++) {
-      layerList[i].hide();
-    }
-    // creates a new layer and add to the stage
-    const layer = new Konva.Layer();
-    stage.add(layer);
-    layerList.push(layer);
-
-    if (is_pair(xs)) {
-      Tree.fromSourceTree(xs).beginDrawingOn(layer).draw(500, 50);
-    } else if (is_function(xs)) {
-      new FunctionTreeNode().drawOnLayer(50, 50, 50, 50, layer);
-    } else {
-      const text = new Konva.Text({
-        text: toText(xs, true),
-        align: 'center',
-        x: 500,
-        y: 50,
-        fontStyle: 'normal',
-        fontSize: 20,
-        fill: 'white'
-      });
-      layer.add(text);
+
+    const panes = [];
+
+    // If this is the first step, create the Konva.Stages used to display the Konva.Layers
+    if (stepList.length == 0)
+      createPanes(structures.length);
+
+    // Create a Pane for each structure
+    for (let i = 0; i < structures.length; i++) {
+      const xs = structures[i]
+      const stage = stages[i];
+
+      /**
+       * Size stage according to calculated width and height of drawing.
+       * Theoretically, as each box is 90px wide and successive boxes overlap by half,
+       * the width of the drawing should be roughly (width * 45), with a similar calculation
+       * for height.
+       * In practice, likely due to browser auto-scaling, for large drawings this results in
+       * some of the drawing being cut off. Hence the width and height formulas used are approximations.
+       */
+      const config = {
+        width: findListWidth(xs) * 60 + 60,
+        height: findListHeight(xs) * 60 + 100,
+      };
+      // creates a new layer and add to the stage
+      const layer = new Konva.Layer();
+
+      if (is_pair(xs)) {
+        Tree.fromSourceTree(xs).beginDrawingOn(layer).draw(500, 50);
+      } else if (is_function(xs)) {
+        new FunctionTreeNode().drawOnLayer(50, 50, 50, 50, layer);
+      } else {
+        const text = new Konva.Text({
+          text: toText(xs, true),
+          align: 'center',
+          x: 500,
+          y: 50,
+          fontStyle: 'normal',
+          fontSize: 20,
+          fill: 'white'
+        });
+        layer.add(text);
+      }
+      // adjust the position
+      layer.offsetX(minLeft - 20);
+      layer.offsetY(0);
+      layer.draw();
+
+      // Show the first structure
+      if (stepList.length == 0) {
```

Comment:
Use ===  instead of == for stricter comparison.

Reply:
Is there a reason for using == instead of === for comparison? 
This rule has not been turned off so eslint is giving me an error on this line. 
https://eslint.org/docs/rules/eqeqeq